#30-34 category module

zim89 opened the pull request.

zim89 committed:
add Category model

- add property CATEGORY to consts/models
- add property COLOR_PATTERN to regex in consts/validation
- add Category model

zim89 committed:
add CRUD for categories

FIX:
 - fixed body check in validationMiddleware

CATEGORY module:
- add create category
- add findAll category
- add findAllById category
- add property color to validationErrors

zim89 committed:
Update category.routes.js

vlad-khrychov commented on docs/category-shema.yaml:
Do not forgot to create task for filling this documentation and for creating tests.

zim89 committed:
add tests for Category module

tests for category model
tests for category controller
tests for category service
